Added InstallAware

msintle opened the pull request:
InstallAware is now included among the InstallShield and WiX third parties.

# InstallAware

This Pull Request (PR) adds information about the InstallAware third party installer, alongside the existing third parties InstallShield and WiX.

## Summary

InstallAware features one-click inclusion of all Microsoft .NET Framework versions in setups it creates. InstallAware's runtime installation mechanism is fully editable and includes scripts which help avoid common installation pitfalls (for example, for the latest Microsoft .NET 4.7 Framework, the InstallAware runtime automatically updates Windows 7 certificates, installs DirectX DLLs, where these would otherwise break the installation with other third parties like InstallShield or WiX).

## Details

InstallAware's runtime installation mechanism features one-click runtime inclusion into setup projects, as well as partial web deployment, and 64-bit compression (maintaining full extraction compatibility on 32-bit systems) which often reduces runtime sizes to 50% of their already-compressed sizes - resulting in an improved user experience and developer experience at the same time.

msintle committed:
Added InstallAware

InstallAware is now included among the InstallShield and WiX third parties.

dnfclas commented:
This seems like a small (but important) contribution, so no Contribution License Agreement is required at this point. We will now review your pull request.
_Thanks,
.NET Foundation Pull Request Bot_

mairaw commented:
Thanks for your contribution @msintle. I sent a request for feedback internally and I'll respond once I hear back.

mairaw commented:
Hi @msintle, I've discussed this with the team.  We're happy to add InstallAware to the topic, however we'd have to remove the marketing tone of the changes and instead do one of the two options:
1. Follow the [WIX](https://github.com/dotnet/docs/blob/master/docs/framework/deployment/deployment-guide-for-developers.md) model, where you give a background about the product and link to the company's page for more info
2. Follow the [InstallShield](https://github.com/dotnet/docs/blob/master/docs/framework/deployment/deployment-guide-for-developers.md#installshield-deployment) model, where you have a how-to/step-by-step guidance.

Thanks! 

/cc @unniravindranathan

msintle commented:
Sounds good. Shall we create a new edit and then submit a new pull request?
Thank you.

On Wed, Aug 16, 2017 at 10:27 AM, Maira Wenzel <notifications@github.com>
wrote:

> Hi @msintle <https://github.com/msintle>, I've discussed this with the
> team. We're happy to add InstallAware to the topic, however we'd have to
> remove the marketing tone of the changes and instead do one of the two
> options:
>
>    1. Follow the WIX
>    <https://github.com/dotnet/docs/blob/master/docs/framework/deployment/deployment-guide-for-developers.md>
>    model, where you give a background about the product and link to the
>    company's page for more info
>    2. Follow the InstallShield
>    <https://github.com/dotnet/docs/blob/master/docs/framework/deployment/deployment-guide-for-developers.md#installshield-deployment>
>    model, where you have a how-to/step-by-step guidance.
>
> Thanks!
>
> /cc @unniravindranathan <https://github.com/unniravindranathan>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/dotnet/docs/pull/2823#issuecomment-322689160>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AdPWHf0FQMxmllhEruz9b2Ic8juyZ7GAks5sYpnsgaJpZM4OryXT>
> .
>

BillWagner commented:
@msintle You can add the new commits on the current branch. When you push the changes to GitHub the PR will be automatically updated with the new changes. There's no need to close this and make a new PR.

msintle committed:
Updated InstallAware Description

Followed guidelines requested by Maria Wenzel in updating the InstallAware description.

msintle commented:
We have followed the WiX example and committed a new set of changes.

WiX Word Count: 71
WiX Link Count: 3

InstallAware Word Count: 67
InstallAware Link Count: 3

Please let us have your thoughts.

On Wed, Aug 16, 2017 at 11:19 AM, Bill Wagner <notifications@github.com>
wrote:

> @msintle <https://github.com/msintle> You can add the new commits on the
> current branch. When you push the changes to GitHub the PR will be
> automatically updated with the new changes. There's no need to close this
> and make a new PR.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/dotnet/docs/pull/2823#issuecomment-322700094>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AdPWHdkr1M1-FtV1jeAOrYFURE5fNovsks5sYqYUgaJpZM4OryXT>
> .
>

mairaw commented on docs/framework/deployment/deployment-guide-for-developers.md:
```diff
@@ -70,8 +70,8 @@ For download links, see the section [Redistributable Packages](#redistributable-
 
```
on line 4, please update ms.date to reflect we're making important changes

msintle replied:
Done.

mairaw commented on docs/framework/deployment/deployment-guide-for-developers.md:
```diff
@@ -137,6 +139,10 @@ If you use ClickOnce, InstallShield, or WiX to deploy your app, you can add a de
 
+<a name="installaware"></a> 
+### InstallAware deployment
+InstallAware builds Windows app (APPX), Windows Installer (MSI), Native Code (EXE), and App-V (Application Virtualization) packages from a single source. Easily [include any version of the .NET Framework](https://www.installaware.com/one-click-pre-requisite-installer.htm) in your setup, optionally customizing the installation by [editing the default scripts](https://www.installaware.com/msicode.h ...
```
remove the for example sentence from the parentheses and put it in a new sentence. Such as:
... in your setup, optionally customizing the installation by [editing the default scripts](https://www.installaware.com/msicode.htm). For example, InstallAware pre-installs certificates on Windows 7, without which the .NET 4.7 setup fails.

Also change .NET 4.7 to .NET Framework 4.7.

mairaw reviewed (changes requested):
Thanks for working on the feedback. I think it's almost good to merge. I've left a few comments to be addressed. 

@unniravindranathan would you like to review this as well? Internal review URL is:
https://review.docs.microsoft.com/en-us/dotnet/framework/deployment/deployment-guide-for-developers?branch=pr-en-us-2823

msintle committed:
ms.date and minor updates

Committing minor updates and the ms.date change requested by @mairaw.

msintle reviewed (commented):
All requested changes committed.

msintle commented:
Thank you for your feedback *@mairaw*. We have committed the changes
requested. Please let us know if this is ready to merge. Thank you.

On Tue, Aug 29, 2017 at 4:44 AM, Maira Wenzel <notifications@github.com>
wrote:

> *@mairaw* requested changes on this pull request.
>
> Thanks for working on the feedback. I think it's almost good to merge.
> I've left a few comments to be addressed.
>
> @unniravindranathan <https://github.com/unniravindranathan> would you
> like to review this as well? Internal review URL is:
> https://review.docs.microsoft.com/en-us/dotnet/framework/
> deployment/deployment-guide-for-developers?branch=pr-en-us-2823
> ------------------------------
>
> In docs/framework/deployment/deployment-guide-for-developers.md
> <https://github.com/dotnet/docs/pull/2823#discussion_r135675943>:
>
> > @@ -70,8 +70,8 @@ For download links, see the section [Redistributable Packages](#redistributable-
>
>
> on line 4, please update ms.date to reflect we're making important changes
> ------------------------------
>
> In docs/framework/deployment/deployment-guide-for-developers.md
> <https://github.com/dotnet/docs/pull/2823#discussion_r135676159>:
>
> > @@ -137,6 +139,10 @@ If you use ClickOnce, InstallShield, or WiX to deploy your app, you can add a de
>
>  8.  In the **Property Pages** dialog box, choose **OK**.
>
> +<a name="installaware"></a>
> +### InstallAware deployment
> +InstallAware builds Windows app (APPX), Windows Installer (MSI), Native Code (EXE), and App-V (Application Virtualization) packages from a single source. Easily [include any version of the .NET Framework](https://www.installaware.com/one-click-pre-requisite-installer.htm) in your setup, optionally customizing the installation by [editing the default scripts](https://www.installaware.com/msicode.htm) (for example, InstallAware pre-installs certificates on Windows 7, without which the .NET 4.7 setup fails). For more information on InstallAware, see the [InstallAware for Windows Installer](https://www.installaware.com/) website.
>
> remove the for example sentence from the parentheses and put it in a new
> sentence. Such as:
> ... in your setup, optionally customizing the installation by editing the
> default scripts <https://www.installaware.com/msicode.htm>. For example,
> InstallAware pre-installs certificates on Windows 7, without which the .NET
> 4.7 setup fails.
>
> Also change .NET 4.7 to .NET Framework 4.7.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/dotnet/docs/pull/2823#pullrequestreview-59112434>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AdPWHXIvs8IUp8-7OuRgGljGnCJ_Ww79ks5sc20UgaJpZM4OryXT>
> .
>

mairaw reviewed (approved):
Thank you @msintle. This is good to merge. I appreciate your patience. I also got approval offline from one of the PMs. It'll go live in our next content push to production, probably later today.
